entrypoint.sh

```diff
@@ -29,12 +43,19 @@ http {
 
   server {
     location / {
-      proxy_pass http://${UPSTREAM}:${UPSTREAM_PORT};
-      proxy_set_header Host \$host;
-      proxy_set_header X-Forwarded-For \$remote_addr;
-      error_page 403 405 414 416 500 501 502 503 504 ${ERROR_PAGE};
-      client_max_body_size ${CLIENT_MAX_BODY_SIZE};
+      include /etc/nginx/directory_proxy.conf;
+    }
+
+    location ^~ /admin/ {
+      include /etc/nginx/directory_proxy.conf;
+      allow ${ADMIN_ALLOW};
+      deny ${ADMIN_DENY};
+    }
+
+    if ($http_x_forwarded_proto != 'https') {
```

Comment:
last time I did something like this, the community saw if statements are the devil in nginx. can you sue a non-if here?